crates/electrum/src/bdk_electrum_client.rs

```diff
@@ -294,15 +301,27 @@ impl<E: ElectrumApi> BdkElectrumClient<E> {
                     match tx_res.height.try_into() {
                         // Returned heights 0 & -1 are reserved for unconfirmed txs.
                         Ok(height) if height > 0 => {
-                            self.validate_merkle_for_anchor(tx_update, tx_res.tx_hash, height)?;
+                            txs_to_validate.push((tx_res.tx_hash, height));
                         }
                         _ => {
                             tx_update.seen_ats.insert((tx_res.tx_hash, start_time));
                         }
                     }
                 }
             }
+
+            if unused_spk_count >= stop_gap {
+                break;
+            }
+        }
+
+        // Batch validate all collected transactions.
+        if !txs_to_validate.is_empty() {
+            let proofs = self.batch_fetch_merkle_proofs(&txs_to_validate)?;
+            self.batch_validate_merkle_proofs(tx_update, proofs)?;
         }
```

Comment:
Instead of having every `populate_with_{}` method call this internally, it will be more efficient and make more logical sense if we extract this so that we only call it at the end of `full_scan` and `sync`.

In other words, `populate_with_{}` should no longer fetch anchors. Instead, they should either mutate, or return a list of `(Txid, BlockId)` for which we try to fetch anchors for in a separate step.

It will be even better if full txs are fetched in a separate step too.

Reply:
@LagginTimes did you figure out why though?